src/main/java/io/cdap/plugin/ToUTF8Action.java

```diff
@@ -179,31 +186,38 @@ public ToUTF8Config(String sourceFilePath, String destFilePath, @Nullable String
     /**
      * Validates the config parameters required for unloading the data.
      */
-    private void validate() throws IllegalArgumentException {
-      try {
-        Charset.forName(charset);
-      } catch (UnsupportedCharsetException e) {
-        throw new IllegalArgumentException("The charset entered is not valid. Please use a value " +
-                     "from https://docs.oracle.com/javase/8/docs/technotes/guides/intl/encoding.doc.html.", e);
+    private void validate(FailureCollector collector) throws ValidationException {
+      if (Strings.isNullOrEmpty(sourceFilePath)) {
+        collector.addFailure("Source file or folder is required.", "Set source file or folder.")
+          .withConfigProperty("sourceFilePath");
+        collector.getOrThrowException();
       }
       try {
-        Pattern.compile(fileRegex);
-      } catch (Exception e) {
-        throw new IllegalArgumentException("The regular expression pattern provided is not a valid " +
-                                             "regular expression.", e);
-      }
-      if (Strings.isNullOrEmpty(sourceFilePath)) {
-        throw new IllegalArgumentException("Source file or folder is required.");
+        Path source = new Path(sourceFilePath);
+        source.getFileSystem(new Configuration());
+      } catch (IOException e) {
+        collector.addFailure("Cannot determine the file system of the source file.", null)
```

Comment:
1. we can't say that all IOExceptions mean the same thing.
2. We lose initial exception that makes debug much harder